15-draft.json

```diff
@@ -1392,7 +1401,22 @@
     "space",
     "logo",
     "url",
-    "location",
     "contact"
+  ],
+  "allOf": [
+    {
+      "if": {
+        "properties": {
+          "type": {
+            "const": "space"
+          }
+        }
+      },
+      "then": {
+        "required": [
+          "location"
+        ]
+      }
+    }
```

Comment:
Conditional validation requirements might be suboptimal, since quite a few tools will probably not support them. However, if a tool ignores the `allOf` key, then location will be optional, which should be fine (as long as people properly use the type field).